core/controllers/admin_test.py

```diff
@@ -1109,18 +1110,74 @@ def test_send_dummy_mail(self):
 
         with self.swap(feconf, 'CAN_SEND_EMAILS', True):
             generated_response = self.post_json(
-                '/sendDummyMailToAdminHandler', payload={},
+                '/senddummymailtoadminhandler', payload={},
                 csrf_token=csrf_token, expected_status_int=200)
             self.assertEqual(generated_response, {})
 
         with self.swap(feconf, 'CAN_SEND_EMAILS', False):
             generated_response = self.post_json(
-                '/sendDummyMailToAdminHandler', payload={},
+                '/senddummymailtoadminhandler', payload={},
                 csrf_token=csrf_token, expected_status_int=400)
             self.assertEqual(
                 generated_response['error'], 'This app cannot send emails.')
 
 
+class UpdateUsernameHandlerTest(test_utils.GenericTestBase):
+    """Tests for updating usernames."""
+
+    def setUp(self):
+        super(UpdateUsernameHandlerTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+
+    def test_update_username(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+
+        current_username, new_username = self.ADMIN_USERNAME, 'newUsername'
+        non_existent_username, long_username = 'invalid', 'a' * 31
+        user_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        # The new username must not be longer than 30 characters.
+        response = self.put_json(
+            '/updateusernamehandler',
+            payload={
+                'current_username': current_username,
+                'new_username': long_username},
+            csrf_token=csrf_token,
+            expected_status_int=400)
+        self.assertEqual(
+            response['error'], 'Please make sure that the new '
+            'username is not longer than %s characters.' % MAX_USERNAME_LENGTH)
+
+        # If current username does not exist.
+        response = self.put_json(
+            '/updateusernamehandler',
+            payload={
+                'current_username': non_existent_username,
+                'new_username': new_username},
+            csrf_token=csrf_token,
+            expected_status_int=400)
+        self.assertEqual(response['error'], 'User does not exist.')
```

Comment:
Create one test for each case. See the[ guidelines in the wiki page](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests)

Reply:
Done.